packages/next/package.json

```diff
@@ -10,66 +10,20 @@
   "types": "index.d.ts",
   "files": [
     "dist",
-    "app.js",
-    "app.d.ts",
-    "babel.js",
-    "babel.d.ts",
-    "client.js",
-    "client.d.ts",
+    "*.js",
+    "*.d.ts",
     "compat",
-    "cache.js",
-    "cache.d.ts",
-    "config.js",
-    "config.d.ts",
-    "constants.js",
-    "constants.d.ts",
-    "document.js",
-    "document.d.ts",
-    "dynamic.js",
-    "dynamic.d.ts",
-    "error.js",
-    "error.d.ts",
+    "experimental",
+    "font",
     "future",
+    "image-types",
     "legacy",
-    "script.js",
-    "script.d.ts",
-    "server.js",
-    "server.d.ts",
-    "head.js",
-    "head.d.ts",
-    "image.js",
-    "image.d.ts",
-    "link.js",
-    "link.d.ts",
-    "router.js",
-    "router.d.ts",
-    "jest.js",
-    "jest.d.ts",
-    "amp.js",
-    "amp.d.ts",
-    "og.js",
-    "og.d.ts",
-    "index.d.ts",
-    "types/index.d.ts",
-    "types/global.d.ts",
-    "types/compiled.d.ts",
-    "image-types/global.d.ts",
-    "navigation-types/navigation.d.ts",
-    "navigation-types/compat/navigation.d.ts",
-    "font",
-    "navigation.js",
-    "navigation.d.ts",
-    "headers.js",
-    "headers.d.ts",
     "navigation-types",
-    "web-vitals.js",
-    "web-vitals.d.ts",
-    "experimental/testmode/playwright.js",
-    "experimental/testmode/playwright.d.ts",
-    "experimental/testmode/playwright/msw.js",
-    "experimental/testmode/playwright/msw.d.ts",
-    "experimental/testmode/proxy.js",
-    "experimental/testmode/proxy.d.ts"
+    "types/compiled.d.ts",
+    "types/global.d.ts",
+    "types/index.d.ts",
+    "!taskfile*",
+    "!webpack*.config.js"
```

Comment:
I think an include list is better than exclude list here so we aren't accidentally publishing configs/build files that shouldn't be.

Reply:
Fair, although arguably we should rather work towards getting rid of as many of the current config/build files as possible, instead of planning on adding more.

I think if we added even more config/build files, at that point, a `config` folder would then be justifiable and we could just ignore a single folder instead. Accidentally publishing a config file - then excluding it in a follow-up release - is probably less of a DX hit than forgetting to include an actual submodule anyway.

Working with the 50+ files contained in this root directory however is increasingly frictional, and these files could technically just be generated anyway, as they are just static aliases to other files that are currently spread out non-systematically. This can be addressed in a follow-up PR though.

This PR aims to be a - maybe somewhat opinionated, but hopefully positive - small DX improvement as a stopgap on the way to properly supporting the standard `exports` field for submodules anyway.

In light of this, feel free to close or merge this PR. :pray:

Reply:
If we could make explicit folders for being published e.g. `public-types` and `internal-types` and then just leave manually specificying top-level files e.g. `server.js` and `server.d.ts` so that we're conscious about adding top-level files then I'd think it'd be fine